cli/script/commands/debug.ts

```diff
@@ -0,0 +1,105 @@
+import * as path from "path";
+import * as Q from "q";
+
+const simctl = require("simctl");
+const which = require("which");
+
+interface IDebugPlatform {
+    getLogProcess(): any;
+}
+
+class AndroidDebugPlatform implements IDebugPlatform {
+    public getLogProcess(): any {
+        try {
+            which.sync("adb");
+        } catch (e) {
+            throw new Error("ADB command not found. Please ensure it is installed and available on your path.");
+        }
+
+        if (!this.isDeviceAvailable()) {
+            throw new Error("No Android devices found. Re-run this command after starting one.");
+        }
+
+        return childProcess.spawn("adb", ["logcat"]);
+    }
+
+    // The following is an example of what the output looks
+    // like when running the "adb devices" command.
+    //
+    // List of devices attached
+    // emulator-5554	device
+    private isDeviceAvailable(): boolean {
+        const output = childProcess.execSync("adb devices").toString();
+        return output.search(/^[\w-]+\s+device$/mi) > -1;
+    }
+}
+
+class iOSDebugPlatform implements IDebugPlatform {
+    private getSimulatorID(): string {
+        const output: any = simctl.list({ devices: true, silent: true });
+        const simulators: string[] = output.json.devices
+                                        .map((platform: any) => platform.devices)
+                                        .reduce((prev: any, next: any) => prev.concat(next))
+                                        .filter((device: any) => device.state === "Booted")
+                                        .map((device: any) => device.id);
+
+        return simulators[0];
+    }
+
+    public getLogProcess(): any {
+        if (process.platform !== "darwin") {
+            throw new Error("iOS debug logs can only be viewed on OS X.");
+        }
+
+        const simulatorID: string = this.getSimulatorID();
+        if (!simulatorID) {
+            throw new Error("No iOS simulators found. Re-run this command after starting one."); 
+        }
+
+        const logFilePath: string = path.join(process.env.HOME, "Library/Logs/CoreSimulator", simulatorID, "system.log");
+        return childProcess.spawn("tail", ["-f", logFilePath]);
+    }
+}
+
+const logMessagePrefix = "[CodePush] ";
+function processLogData(logData: Buffer) {
+    const content = logData.toString()
+    content.split("\n")
+        .filter((line: string) => line.indexOf(logMessagePrefix) > -1)
+        .map((line: string) => {
+            const timeStamp = moment().format("hh:mm:ss");
+            const message = line.substring(line.indexOf(logMessagePrefix) + logMessagePrefix.length);
+            return `[${timeStamp}] ${message}`;
+        })
+        .forEach((line: string) => console.log(line));
+}
+
+const debugPlatforms: any = {
+    android: new AndroidDebugPlatform(),
+    ios: new iOSDebugPlatform()
+};
+
+module.exports = (command: cli.IDebugCommand): Q.Promise<void> => {
```

Comment:
Given that we are not moving this into the command-executor, may be change this to be an `export function debugCommand(command: cli.IDebugCommand): Q.Promise<void> {...}` to be consistent with the exports in our ts code and use ` import { debugCommand } from "./commands/debug" ` in command-executor.